apps/bridge-dapp/src/containers/DeleteNotesModal/DeleteNotesModal.tsx

```diff
@@ -38,13 +38,16 @@ export const DeleteNotesModal: FC<DeleteNotesModalProps> = ({
     }
   }, [notes, setNotes]);
 
-  const handleDeleteNotes = useCallback(() => {
+  const handleDeleteNotes = useCallback(async () => {
     if (!notes || !noteManager) {
       console.trace('No notes or note manager found');
       return;
     }
 
-    notes.forEach((note) => noteManager.removeNote(note));
+    for (const note of notes) {
+      await noteManager.removeNote(note);
+    }
+
```

Comment:
You can use `Promise.all` or `Promise.allSettled` here for remove note concurrently and consider wrapping the code in the try/catch block for better error handling.